fastcrypto-tbls/src/tbls.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Some of the code below is based on code from https://github.com/celo-org/celo-threshold-bls-rs,
+// modified for our needs.
+
+use fastcrypto::error::FastCryptoError;
+use fastcrypto::groups::{bls12381, GroupElement, HashToGroupElement, Pairing, Scalar};
+use fastcrypto::polynomial::{IndexedValue, Poly};
+
+pub type Share<S> = IndexedValue<S>;
+pub type PartialSignature<S> = IndexedValue<S>;
+
+/// Trait [ThresholdBls] provides sign & verify functions for standard and partial BLS signatures.
+pub trait ThresholdBls {
+    type Private: Scalar;
+    /// `Public` represents the group over which the public keys are represented.
+    type Public: GroupElement<ScalarType = Self::Private>;
+    /// `Signature` represents the group over which the signatures are represented.
+    type Signature: GroupElement<ScalarType = Self::Private> + HashToGroupElement;
+
+    fn verify_pairings(p: &Self::Public, sig: &Self::Signature, hm: &Self::Signature) -> bool;
+
+    fn sign(private: &Self::Private, msg: &[u8]) -> Self::Signature {
```

Comment:
For signing I guess we're fine and will never fail? iirc we have `try_sign` returning a `Result` in the rest of fastcrypto.

Reply:
I don't a case we miss here

Reply:
we use both Signature and Result<Signature, Error> in sui/fastcrypto. I think the Signature without error is more appropriate since there is really no valid error cases here?

Reply:
We're good if signing can never fail.